executor/runtime/types/container.go

```diff
@@ -148,6 +148,10 @@ type TitusInfoContainer struct {
 	logUploaderConfig           uploader.Config
 
 	pod *corev1.Pod
+	// extraUserContainers stores and array of metadata about all the non-main user containers
+	extraUserContainers []*ExtraContainer
+	// extraUserContainers stores and array of metadata about all the platform-defined containers
```

Comment:
Hmm, do you think maybe I should do a map here so I can more naturally iterate over these by name, and look up one of their healths by name?

Reply:
Maybe, though I'd assume YAGNI for now and refactor that as part of the healthcheck work, if it makes sense then.